File: restring/src/test/java/com/ice/restring/ViewTransformerManagerTest.java
package com.ice.restring;

import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mockito;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;

import static org.junit.Assert.assertSame;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.doReturn;
import static org.mockito.Mockito.when;

@RunWith(RobolectricTestRunner.class)
public class ViewTransformerManagerTest {

    private ViewTransformerManager transformerManager;

    @Before
    public void setUp() {
        transformerManager = new ViewTransformerManager();
    }

    @Test
    public void shouldTransformView() {
        TextView textView = new TextView(RuntimeEnvironment.application);

        ViewTransformerManager.Transformer transformer = Mockito.mock(ViewTransformerManager.Transformer.class);
        doReturn(TextView.class).when(transformer).getViewType();
        when(transformer.transform(any(), any())).thenReturn(textView);
        transformerManager.registerTransformer(transformer);

        View transformedView = transformerManager.transform(
                new TextView(RuntimeEnvironment.application),
                Mockito.mock(AttributeSet.class)
        );

        assertSame(textView, transformedView);
    }
}

File: restring/src/test/java/com/b3nedikt/restring/ViewTransformerManagerTest.kt
package com.b3nedikt.restring

import android.widget.TextView
import androidx.test.core.app.ApplicationProvider
import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import org.junit.Assert.assertSame
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class ViewTransformerManagerTest {

    private lateinit var transformerManager: ViewTransformerManager

    @Before
    fun setUp() {
        transformerManager = ViewTransformerManager()
    }

    @Test
    fun shouldTransformView() {
        val textView = TextView(ApplicationProvider.getApplicationContext())

        val transformer = mock<ViewTransformer<TextView>>()

        whenever(transformer.viewType).thenReturn(TextView::class.java)

        transformerManager.registerTransformer(transformer)

        val transformedView = transformerManager.transform(textView, mock())

        assertSame(textView, transformedView)
    }
}